mountpoint-s3/src/prefetch.rs

```diff
@@ -250,29 +277,53 @@ where
 
         let request_task = {
             let client = Arc::clone(&self.inner.client);
+            let preferred_part_size = self.preferred_part_size;
             let part_queue = Arc::clone(&part_queue);
             let bucket = self.bucket.to_owned();
             let key = self.key.to_owned();
             let etag = self.etag.clone();
             let span = debug_span!("prefetch", range=?range);
 
             async move {
+                // CRC32C checksum of the data from get object result
+                // TODO get this value from S3 API if possible
+                let mut get_object_checksum: ChecksumCrc = 0;
+                let mut parts_checksum: ChecksumCrc = 0;
                 match client.get_object(&bucket, &key, Some(range.clone()), Some(etag)).await {
                     Err(e) => {
                         error!(error=?e, "RequestTask get object failed");
-                        part_queue.push(Err(e));
+                        part_queue.push(Err(e.into()));
                     }
                     Ok(request) => {
                         pin_mut!(request);
                         loop {
                             match request.next().await {
                                 Some(Ok((offset, body))) => {
```

Comment:
I wonder if we should instead do `let body = Bytes::from(body);` here, so that the splitting into ranges stuff doesn't actually have to copy the underlying bytes.